sdk/client/generic_swagger_error.go

```diff
@@ -13,9 +13,10 @@ import "fmt"
 
 // GenericSwaggerError Provides access to the body, error and model on returned errors.
 type GenericSwaggerError struct {
-	body  []byte
-	error string
-	model interface{}
+	body       []byte
+	error      string
+	model      interface{}
+	statusCode int
```

Comment:
The `error` string sometimes is the status code and sometimes it's the body of the response

`statusCode` will expose the HTTP status code, always.